### ISSUE TITLE
Add Stackdriver integration

### ISSUE THREAD

**Issue by douglas-reid** [istio/mixer#90]
Tracking issue.

Provide adapters for Stackdriver logging/monitoring.  This will require setup of creds passing in config, etc.

**Comment by rshriram** [comment]
Should cloud platform specific adapters go into this mixer repo or should it be moved into a separate contrib repo? Stackdriver is just one logging and monitoring option. There are a ton of things out there. 

I would suggest we add adapter interface for logging and monitoring, and provide a reference adapter for say elasticsearch (or something simpler) and Prometheus. 
The idea being folks can try out all aspects of istio functionality on a local dev box with minimal overhead. 

Platform specific adapters could go into a contrib repo under mixer/adapters/logging/ (stackdriver, splunk, hosted elasticsearch, etc). These would certainly need credentials etc. 

Thoughts ?

**Comment by douglas-reid** [comment]
Where the adapter code lives should not impact the ability of anyone to run a mixer server on a local dev box. I'm not sure the overhead of compiling in the extra code is worth worrying about (at this point) -- and the adapter won't be activated unless configured.

I suggest we start with adding adapters of all kinds to mixer proper (a la heapster) and if that gets to be overwhelming or problematic, then we split it out.

FWIW, none of these issues have priorities or milestones yet, so this issue is only intended to capture the fact that, at some point, we should add the ability for mixer to export logs and metrics to Stackdriver (among a wide variety of others).

**Comment by wora** [comment]
One critical issue is the schema. Google APIs already send data to
Stackdriver. We need to decide whether to use identical schema or separate
one. There is a major refactoring pending on this topic, which will affect
this effort.

**Comment by douglas-reid** [comment]
@wora to clarify, are you talking about identical metrics schema (metric names and label names, etc.) in Stackdriver?

I believe the answer there is that the schema should be defined in the config for the stackdriver adapter.  If that proves too unwieldy, we can define a new Aspect for Stackdriver generic and maybe Stackdriver googleapis?

@mandarjog for comment.

**Comment by ZackButcher** [comment]
MVP was shipped, we can open up new issues for further work.
